packages/json/src/node/JsonAstNode.ts

```diff
@@ -156,9 +156,10 @@ export namespace JsonStringNode {
 export interface JsonNumberExpectation extends JsonBaseExpectation {
 	readonly type: 'json:number'
 }
-export interface JsonNumberNode extends core.FloatBaseNode, JsonBaseAstNode {
+export interface JsonNumberNode extends JsonBaseAstNode, core.AstNode {
 	readonly type: 'json:number'
-	readonly value: number
+	readonly children: [core.LongNode | core.FloatNode]
```

Comment:
Does this need to be a `LongNode`? Wouldn't `IntegerNode` be enough, because this is JSON?

Reply:
I used long node because it uses a bigint internally, so the parsed value can be more precise.

Reply:
That's what I mean. Does it need that precision? Aren't JSON values inherently restricted by the precision of JS numbers?

Reply:
I just figured there isn't any disadvantage of using a bigint. Not sure what the json spec says, and also not sure how Mojang handles this when codec specifies a long.

Reply:
Fair. I think bigints are a bit annoying in JS, for example if you have `int @ 1..2` and you parse the `1..2` range as numbers but the json number itself is a bigint, you need to convert one to the other to compare them. Either way no major issues so I'm fine with it.